src/cmd/flux-resource.py

```diff
@@ -20,6 +20,58 @@
 from flux.idset import IDset
 from flux.resource import ResourceSet, SchedResourceList, resource_list
 from flux.rpc import RPC
+from flux.util import UtilConfig
+
+
+class FluxResourceConfig(UtilConfig):
+    """flux-resource specific user configuration class"""
+
+    builtin_formats = {}
+    builtin_formats["status"] = {
+        "default": {
+            "description": "Default flux-resource status format string",
+            "format": "{state:>10} {nnodes:>6} {nodelist}",
+        },
+    }
+    builtin_formats["drain"] = {
+        "default": {
+            "description": "Default flux-resource drain format string",
+            "format": (
+                "{timestamp!d:%FT%T::<20} {state:<8.8} {ranks:<8.8+} "
+                "{reason:<30.30+} {nodelist}"
+            ),
+        },
+    }
+    builtin_formats["list"] = {
+        "default": {
+            "description": "Default flux-resource list format string",
+            "format": (
+                "{state:>10} ?:{properties:<10.10+} {nnodes:>6} "
+                "{ncores:>8} {ngpus:>8} {nodelist}"
+            ),
+        },
+    }
+
+    def __init__(self, subcommand):
+        initial_dict = {}
+        for key, value in self.builtin_formats.items():
+            initial_dict[key] = {"formats": value}
+        super().__init__(
+            name="flux-resource", subcommand=subcommand, initial_dict=initial_dict
+        )
+
+    def validate(self, path, conf):
+        """Validate a loaded flux-resource config file as dictionary"""
+
+        for key, value in conf.items():
+            if key in ["status", "list", "drain"]:
+                for key2, val2 in value.items():
+                    if key2 == "formats":
+                        self.validate_formats(path, val2)
+                    else:
+                        raise ValueError(f"{path}: invalid key {key}.{key2}")
+            else:
+                raise ValueError(f"{path}: invalid key {key}")
```

Comment:
would subcommand fail to run if a config error exists for a different subcommand?  Maybe only want to check the subcommand that user chose?

Reply:
I thought about this, but thought it would be less annoying to validate the whole config file for the `flux-resource` command as it is loaded, so that a user doesn't have to run 3 separate subcommands to get validation.